easybuild/framework/easyconfig/easyconfig.py

```diff
@@ -549,7 +551,8 @@ def _parse_dependency(self, dep, hidden=False):
             'version': None,
             'versionsuffix': '',
             'hidden': hidden,
-            'external_module': False
+            'external_module': False,
+            'root': None,
```

Comment:
add a comment to explain what this is about (and maybe also the ones above). it seems like magic here